### PR TITLE
[WIP] adding syntaxDecl model

### PR DESCRIPTION
**Pull request by Emanon42**

### CODE REVIEW

File: View/SyntaxDecl.hs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module View.SyntaxDecl where
+import Miso
+import qualified Miso.String as MS
+import qualified SyntaxDecl as SD
+import View.Term
+import View.Utils
+
+renderSyntaxDecl i opts tbl textIn selected (SD.SyntaxDecl n p a tbl2) = div_ []
```
Comment (Emanon42) [inline View/SyntaxDecl.hs:10]: I think the bug is in this function but I have no idea about how to write renderSyntaxDecl in correct way.

File: SyntaxDecl.hs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  True -> pure ()
+  False -> errorMessage "Precedence must be integer"
+  where
+    go :: MS.MisoString -> [Bool]
+    go prec = case MS.uncons prec of
+      Just (x, "") -> [isDigit x]
+      Just (x, xs) -> (isDigit x) : go xs
+
+instance Control SyntaxDecl where
+  data Focus SyntaxDecl = NewSyntaxDeclFocus
```
Comment (liamoc) [inline SyntaxDecl.hs:56]: You probably only need one focus for syntaxdecl, to edit the name. Changing associativity and precedence would be only actions

File: SyntaxDecl.hs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                        | NameFocus
+                        | PrecFocus
+                        | AssocFocus
+                        deriving (Show, Eq)
+  data Action SyntaxDecl = AddSyntaxDecl
```
Comment (liamoc) [inline SyntaxDecl.hs:61]: The focus and action to add a new syntaxdecl would be outside the syntaxdecl and part of editor.

File: View/SyntaxDecl.hs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  $ (block "item-syntaxdecl" [anchor i ["SyntaxDecl."]])
+  : button "editable editable-name" "" (SetFocus SD.NameFocus) [anchor i [text n]]
+  : button "editable editable-prec" "" (SetFocus SD.PrecFocus) [anchor i [text $ MS.toMisoString $ show p]]
```
Comment (liamoc) [inline View/SyntaxDecl.hs:13]: For the precedence, i would suggest using input tags (type = number), for associativity i would use radio buttons or a combobox. Then you can just put an action on them directly and don't need a focus.

For the name, use the `editor` widget (from `View.Utils`)  which you should only display if the current focus is on that item. See the `editableMath` widget from `View.Utils` for an example of how to do this.

**Pull request closed without merging**